File: src/main/java/com/example/chatService/dto/OnlineUser.java
package com.example.chatService.dto;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class OnlineUser {
    private String userId;
    private String username;

    public OnlineUser() {}

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof OnlineUser)) return false;
        OnlineUser that = (OnlineUser) o;
        return userId.equals(that.userId);
    }

    @Override
    public int hashCode() {
        return userId.hashCode();
    }
}

File: src/main/java/com/example/chatService/dto/OnlineUser.kt
package com.example.chatService.dto

data class OnlineUser(
    var userId: String = "",
    var username: String = ""
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is OnlineUser) return false
        return userId == other.userId
    }

    override fun hashCode(): Int = userId.hashCode()
}
